tests/e2e/pageobjects/dashboard/Dashboard.ts

```diff
@@ -63,11 +63,29 @@ export class Dashboard {
     }
 
     async waitPage(timeout: number = TestConstants.TS_SELENIUM_LOAD_PAGE_TIMEOUT) {
+        const polling: number = TestConstants.TS_SELENIUM_DEFAULT_POLLING;
+        const attempts: number = Math.ceil(timeout / polling);
         Logger.debug('Dashboard.waitPage');
 
-        await this.driverHelper.waitVisibility(By.css(Dashboard.WORKSPACES_BUTTON_CSS), timeout);
-        await this.driverHelper.waitVisibility(By.css(Dashboard.STACKS_BUTTON_CSS), timeout);
-        await this.driverHelper.waitVisibility(By.xpath(Dashboard.GET_STARTED_BUTTON_XPATH), timeout);
+        let workspacesLocated: boolean = false;
+        let stacksLocated: boolean = false;
+        for (let i = 0; i < attempts; i++)  {
+            try {
+                if (!workspacesLocated) { await this.driverHelper.waitVisibility(By.css(Dashboard.WORKSPACES_BUTTON_CSS), polling); }
+                workspacesLocated = true;
+                if (!stacksLocated) { await this.driverHelper.waitVisibility(By.css(Dashboard.STACKS_BUTTON_CSS), polling); }
+                stacksLocated = true;
+                await this.driverHelper.waitVisibility(By.xpath(Dashboard.GET_STARTED_BUTTON_XPATH), polling);
+                return;
+            } catch (err) {
+                if (i >= attempts) {
+                    Logger.error(`Dashboard.waitPage - failed with exception, out of attempts - ${err}`);
+                    throw err;
+                } else {
```

Comment:
I guess you should continue only in case if an error is "TimeoutError". If error has any other type it means that something went wrong and such error should be thrown outside.

Reply:
How about stale element, is it a member of TimeoutError?

Reply:
fixed